cmd/zoekt-sourcegraph-indexserver/index.go

```diff
@@ -321,6 +324,34 @@ func gitIndex(c gitIndexConfig, o *indexArgs, l sglog.Logger) error {
 		"-submodules=false",
 	}
 
+	// We store the document ranks as JSON in gitDir and add a flag to tell
+	// zoekt-git-index to rank based on the scores in the file, instead of ranking
+	// the files by itself.
+	if o.UseOfflineRanking {
+		args = append(args, "-offline_ranking")
+
+		r, err := sourcegraph.GetDocumentRanks(context.Background(), o.Name)
+		if err != nil {
+			return err
+		}
+
+		b, err := json.Marshal(r)
+		if err != nil {
+			return err
+		}
+
+		fd, err := os.Create(filepath.Join(gitDir, zoekt.DocumentScoresFile))
+		if err != nil {
+			return err
+		}
+		defer fd.Close()
```

Comment:
this defer will run when the function is done, which means there is data that may not have flushed to disk when you cal zoekt-git-index. I recommend using `os.WriteFile` instead.